pkgs/by-name/po/pocket-id/package.nix

```
@@ -36,22 +38,28 @@ buildGoModule (finalAttrs: {
     mv $out/bin/cmd $out/bin/pocket-id
   '';
 
-  frontend = buildNpmPackage {
+  frontend = stdenv.mkDerivation {
     pname = "pocket-id-frontend";
     inherit (finalAttrs) version src;
 
-    sourceRoot = "${finalAttrs.src.name}/frontend";
-
-    npmDepsHash = "sha256-FiFSnN6DOMr8XghvyGTWB/EMTNfvpqlAgx7FPnbGQxU=";
-    npmFlags = [ "--legacy-peer-deps" ];
+    nativeBuildInputs = [
+      nodejs
+      pnpm_10.configHook
+    ];
+    pnpmDeps = pnpm_10.fetchDeps {
+      inherit (finalAttrs) pname version src;
+      fetcherVersion = 1;
+      hash = "sha256-UgbclnoOqsWY5fYAGoDJON9MDtN5edw65JRleghdReE=";
+    };
 
     env.BUILD_OUTPUT_PATH = "dist";
 
     installPhase = ''
       runHook preInstall
 
+      pnpm --filter pocket-id-frontend build
```

Nit: building the frontend should go into its own phase
```suggestion
    buildPhase = ''
      runHook preBuild

      pnpm --filter pocket-id-frontend build

      runHook postBuild
    '';

    installPhase = ''
      runHook preInstall

```